### PR TITLE
Quick-reconnect when user switches network

### PR DESCRIPTION
**Pull request by hiroshihorie**
Previously, switching network doesn't initiate a quick-reconnect (or the user needs to wait until PeerConnectionState becomes `failed`)

This change instantly resumes the session by quick-reconnecting to improve user experience.

### CODE REVIEW

File: Sources/LiveKit/Extensions/TimeInterval.swift [hunk 1]
```diff
@@ -3,7 +3,8 @@ import Foundation
 /// Default timeout `TimeInterval`s used throughout the SDK.
 internal extension TimeInterval {
     static let captureStart: Self = 5
-    static let defaultConnect: Self = 10
+    static let defaultConnect: Self = 5
```
Comment (boks1971) [inline Sources/LiveKit/Extensions/TimeInterval.swift:6]: Is this the connect timeout? We have seen that this is not enough on long delay networks? We are using 15 seconds in JS SDK for connection timeout.

File: Sources/LiveKit/Support/ConnectivityListener.swift [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        }
+    }
+
+    public private(set) var path: NWPath {
+        didSet {
+            defer { self.hasConnectivity = path.hasConnectivity() }
+
+            if oldValue.hasConnectivity(), path.hasConnectivity(),
+               // first interface changed
+               (oldValue.availableInterfaces.first != path.availableInterfaces.first)
```
Comment (boks1971) [inline Sources/LiveKit/Support/ConnectivityListener.swift:40]: Is `first` the active one always?

File: Sources/LiveKit/Support/ConnectivityListener.swift [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                // same interface but gateways changed (detect wifi network switch)
+                || (oldValue.gateways != path.gateways) {
```
Comment (boks1971) [inline Sources/LiveKit/Support/ConnectivityListener.swift:42]: Is this a `route` change? When does this happen?

Is it possible to detect IP address change? I guess if it is NATted it is not possible. To keep it simple, I guess letting any IP address change take the ICE failed route is reasonable.

**Review by boks1971 — APPROVED** [review]
lgtm! but I barely under `swift` code :-)

**Review by davidzhao — APPROVED** [review]
it's great to have this!

File: Sources/LiveKit/Support/ConnectivityListener.swift (hunk 2 above)
Comment (hiroshihorie) [reply Sources/LiveKit/Support/ConnectivityListener.swift:40]: https://developer.apple.com/documentation/network/nwpath/2998720-availableinterfaces
"in order of preference" 

yes it should be !

**Pull request merged**